File: py/features.py
# DigimonTcg SDK feature factory

from feature.base_feature import DigimonTcgBaseFeature
from feature.test_feature import DigimonTcgTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DigimonTcgBaseFeature(),
        "test": lambda: DigimonTcgTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()

File: rb/features.rb
# DigimonTcg SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module DigimonTcgFeatures
  def self.make_feature(name)
    case name
    when "base"
      DigimonTcgBaseFeature.new
    when "ratelimit"
      DigimonTcgRatelimitFeature.new
    when "retry"
      DigimonTcgRetryFeature.new
    when "test"
      DigimonTcgTestFeature.new
    when "timeout"
      DigimonTcgTimeoutFeature.new
    else
      DigimonTcgBaseFeature.new
    end
  end
end
